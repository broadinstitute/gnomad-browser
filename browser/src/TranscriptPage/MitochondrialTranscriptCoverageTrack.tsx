import React from 'react'

import {
  DatasetId,
  labelForDataset,
  referenceGenome,
  hasMitochondrialGenomeCoverage,
} from '@gnomad/dataset-metadata/metadata'
import CoverageTrack from '../CoverageTrack'
import Query from '../Query'
import StatusMessage from '../StatusMessage'

const operationName = 'MitochondrialCoverageInTranscript'
const query = `
query ${operationName}($transcriptId: String!, $datasetId: DatasetId!, $referenceGenome: ReferenceGenomeId!) {
  transcript(transcript_id: $transcriptId, reference_genome: $referenceGenome) {
    mitochondrial_coverage(dataset: $datasetId) {
      pos
      mean
      median
      over_100
      over_1000
    }
  }
}
`

type Props = {
  datasetId: DatasetId
  transcriptId: string
}

const MitochondrialTranscriptCoverageTrack = ({ datasetId, transcriptId }: Props) => {
  if (!hasMitochondrialGenomeCoverage(datasetId)) {
    return (
      <StatusMessage>
        Mitochondrial genome coverage is not available in {labelForDataset(datasetId)}
      </StatusMessage>
    )
  }

  return (
    <Query
      operationName={operationName}
      query={query}
      variables={{ transcriptId, datasetId, referenceGenome: referenceGenome(datasetId) }}
      loadingMessage="Loading coverage"
      loadingPlaceholderHeight={220}
      errorMessage="Unable to load coverage"
      success={(data: any) => data.transcript && data.transcript.mitochondrial_coverage}
    >
      {({ data }: any) => {
        const coverage = [
          {
            color: 'rgb(115, 171, 61)',
            buckets: data.transcript.mitochondrial_coverage,
            name: 'mitochondrial genome',
            opacity: 0.7,
          },
        ]

        return (
          // @ts-expect-error TS(2769) FIXME: No overload matches this call.
          <CoverageTrack
            coverageOverThresholds={[100, 1000]}
            datasets={coverage}
            filenameForExport={() => `${transcriptId}_coverage`}
            height={190}
            maxCoverage={3000}
            datasetId={datasetId}
          />
        )
      }}
    </Query>
  )
}

export default MitochondrialTranscriptCoverageTrack
