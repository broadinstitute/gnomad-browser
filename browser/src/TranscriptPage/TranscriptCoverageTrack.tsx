import React from 'react'

import { referenceGenome } from '@gnomad/dataset-metadata/metadata'
import { coverageDataset } from '../coverage'
import { coverageConfigClassic, coverageConfigNew } from '../coverageStyles'
import CoverageTrack from '../CoverageTrack'
import Query from '../Query'

const operationName = 'TranscriptCoverage'
const coverageQuery = `
query ${operationName}($transcriptId: String!, $datasetId: DatasetId!, $referenceGenome: ReferenceGenomeId!, $includeExomeCoverage: Boolean!, $includeGenomeCoverage: Boolean!) {
  transcript(transcript_id: $transcriptId, reference_genome: $referenceGenome) {
    coverage(dataset: $datasetId) {
      exome @include(if: $includeExomeCoverage) {
        pos
        mean
        median
        over_1
        over_5
        over_10
        over_15
        over_20
        over_25
        over_30
        over_50
        over_100
      }
      genome @include(if: $includeGenomeCoverage) {
        pos
        mean
        median
        over_1
        over_5
        over_10
        over_15
        over_20
        over_25
        over_30
        over_50
        over_100
      }
    }
  }
}
`

type OwnProps = {
  datasetId: string
  transcriptId: string
  includeExomeCoverage?: boolean
  includeGenomeCoverage?: boolean
}

// @ts-expect-error TS(2456) FIXME: Type alias 'Props' circularly references itself.
type Props = OwnProps & typeof TranscriptCoverageTrack.defaultProps

// @ts-expect-error TS(7022) FIXME: 'TranscriptCoverageTrack' implicitly has type 'any... Remove this comment to see the full error message
const TranscriptCoverageTrack = ({
  datasetId,
  transcriptId,
  includeExomeCoverage,
  includeGenomeCoverage,
}: Props) => {
  return (
    <Query
      operationName={operationName}
      query={coverageQuery}
      variables={{
        transcriptId,
        datasetId: coverageDataset(datasetId),
        referenceGenome: referenceGenome(coverageDataset(datasetId)),
        includeExomeCoverage,
        includeGenomeCoverage,
      }}
      loadingMessage="Loading coverage"
      loadingPlaceholderHeight={220}
      errorMessage="Unable to load coverage"
      success={(data: any) => {
        if (!data.transcript || !data.transcript.coverage) {
          return false
        }
        const exomeCoverage = includeExomeCoverage ? data.transcript.coverage.exome : true
        const genomeCoverage = includeGenomeCoverage ? data.transcript.coverage.genome : true
        return exomeCoverage || genomeCoverage
      }}
    >
      {({ data }: any) => {
        const exomeCoverage = includeExomeCoverage ? data.transcript.coverage.exome : null
        const genomeCoverage = includeGenomeCoverage ? data.transcript.coverage.genome : null

        const coverageConfig =
          datasetId === 'exac'
            ? coverageConfigClassic(exomeCoverage, genomeCoverage)
            : coverageConfigNew(exomeCoverage, genomeCoverage)

        return (
          // @ts-expect-error TS(2769) FIXME: No overload matches this call.
          <CoverageTrack
            coverageOverThresholds={[1, 5, 10, 15, 20, 25, 30, 50, 100]}
            datasets={coverageConfig}
            filenameForExport={() => `${transcriptId}_coverage`}
            height={190}
          />
        )
      }}
    </Query>
  )
}

TranscriptCoverageTrack.defaultProps = {
  includeExomeCoverage: true,
  includeGenomeCoverage: true,
}

export default TranscriptCoverageTrack
