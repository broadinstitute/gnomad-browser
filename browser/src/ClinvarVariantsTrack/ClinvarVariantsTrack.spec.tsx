import { describe, expect, test } from '@jest/globals'
import 'jest-styled-components'
import clinvarVariantFactory from '../__factories__/ClinvarVariant'
import { ClinvarVariant } from '../VariantPage/VariantPage'
import React from 'react'
import userEvent from '@testing-library/user-event'
import ClinvarVariantTrack from './ClinvarVariantTrack'
import { Transcript } from '../TranscriptPage/TranscriptPage'
import transcriptFactory from '../__factories__/Transcript'
import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'
// @ts-expect-error TS(7016) FIXME: Could not find a declaration file for module '@gno... Remove this comment to see the full error message
import { RegionViewerContext } from '@gnomad/region-viewer'
import { BrowserRouter } from 'react-router-dom'

describe('Clinvar Variants Track', () => {
  const mockClinvarVariants: ClinvarVariant[] = [
    clinvarVariantFactory.build({ gold_stars: 0, major_consequence: 'missense_variant' }),
    clinvarVariantFactory.build({ gold_stars: 1, major_consequence: 'missense_variant' }),
    clinvarVariantFactory.build({ gold_stars: 2, major_consequence: 'missense_variant' }),
    clinvarVariantFactory.build({ gold_stars: 3, major_consequence: 'missense_variant' }),
    clinvarVariantFactory.build({ gold_stars: 4, major_consequence: 'missense_variant' }),
  ]

  const mockTranscripts: Transcript[] = [
    transcriptFactory.build(),
    transcriptFactory.build(),
    transcriptFactory.build(),
    transcriptFactory.build(),
  ]

  const childProps = {
    centerPanelWidth: 3,
    isPositionDefined: true,
    leftPanelWidth: 4,
    regions: [],
    rightPanelWidth: 5,
    scalePosition: (i: number) => i,
  }

  test('renders correctly with default props', () => {
    const tree = renderer.create(
      <RegionViewerContext.Provider value={childProps}>
        <ClinvarVariantTrack
          referenceGenome="GRCh38"
          transcripts={mockTranscripts}
          variants={mockClinvarVariants}
        />
      </RegionViewerContext.Provider>
    )
    expect(tree).toMatchSnapshot()
  })

  test('Allow user to change to different review status filters', async () => {
    const user = userEvent.setup()
    render(
      <BrowserRouter>
        <RegionViewerContext.Provider value={childProps}>
          <ClinvarVariantTrack
            referenceGenome="GRCh38"
            transcripts={mockTranscripts}
            variants={mockClinvarVariants}
          />
        </RegionViewerContext.Provider>
      </BrowserRouter>
    )

    const filterSelect = screen.getByRole('combobox')
    const allStarOpt = screen.getByRole('option', { name: '0-4 Stars' }) as HTMLOptionElement
    const OnePlusStarOpt = screen.getByRole('option', { name: '>=1 Stars' }) as HTMLOptionElement
    const TwoPlusStarOpt = screen.getByRole('option', { name: '>=2 Stars' }) as HTMLOptionElement
    const ThreePlusStarOpt = screen.getByRole('option', { name: '>=3 Stars' }) as HTMLOptionElement
    const FourStarOpt = screen.getByRole('option', { name: '4 Stars' }) as HTMLOptionElement

    expect(allStarOpt.selected).toBe(true)

    await user.selectOptions(filterSelect, OnePlusStarOpt)

    expect(OnePlusStarOpt.selected).toBe(true)
    expect(allStarOpt.selected).toBe(false)

    await user.selectOptions(filterSelect, TwoPlusStarOpt)

    expect(TwoPlusStarOpt.selected).toBe(true)
    expect(OnePlusStarOpt.selected).toBe(false)

    await user.selectOptions(filterSelect, ThreePlusStarOpt)

    expect(ThreePlusStarOpt.selected).toBe(true)
    expect(TwoPlusStarOpt.selected).toBe(false)

    await user.selectOptions(filterSelect, FourStarOpt)

    expect(FourStarOpt.selected).toBe(true)
    expect(ThreePlusStarOpt.selected).toBe(false)
  })

  test('review status selector filters correctly ', async () => {
    const user = userEvent.setup()
    render(
      <BrowserRouter>
        <RegionViewerContext.Provider value={childProps}>
          <ClinvarVariantTrack
            referenceGenome="GRCh38"
            transcripts={mockTranscripts}
            variants={mockClinvarVariants}
          />
        </RegionViewerContext.Provider>
      </BrowserRouter>
    )
    const filterSelect = screen.getByRole('combobox')

    expect(screen.getByText('ClinVar variants (5)')).not.toBeNull()
    await user.selectOptions(
      filterSelect,
      screen.getByRole('option', { name: '>=1 Stars' }) as HTMLOptionElement
    )
    expect(screen.getByText('ClinVar variants (4)')).not.toBeNull()
    await user.selectOptions(
      filterSelect,
      screen.getByRole('option', { name: '>=2 Stars' }) as HTMLOptionElement
    )
    expect(screen.getByText('ClinVar variants (3)')).not.toBeNull()
    await user.selectOptions(
      filterSelect,
      screen.getByRole('option', { name: '>=3 Stars' }) as HTMLOptionElement
    )
    expect(screen.getByText('ClinVar variants (2)')).not.toBeNull()
    await user.selectOptions(
      filterSelect,
      screen.getByRole('option', { name: '4 Stars' }) as HTMLOptionElement
    )
    expect(screen.getByText('ClinVar variants (1)')).not.toBeNull()
  })
})
