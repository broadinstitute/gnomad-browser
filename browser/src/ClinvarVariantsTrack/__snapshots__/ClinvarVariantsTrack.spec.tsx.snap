// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clinvar Variants Track renders correctly with default props 1`] = `
[
  .c26 {
  box-sizing: border-box;
  height: calc(2em + 2px);
  padding: 0.375em 0.75em;
  border-color: #6c757d;
  border-style: solid;
  border-width: 1px;
  border-radius: 0.5em;
  background-color: #f8f9fa;
  cursor: pointer;
  font-size: 1em;
  line-height: 1.25;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c26:active,
.c26:hover {
  background-color: #cbd3da;
}

.c26:disabled {
  background-color: rgba(248,249,250,0.5);
  cursor: not-allowed;
}

.c26:focus {
  box-shadow: 0 0 0 0.2em rgba(108,117,125,0.5);
}

.c26 svg {
  position: relative;
  top: 0.11em;
  width: 0.9em;
  height: 0.9em;
}

.c23 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c24 {
  margin-right: 0.5em;
}

.c28 {
  color: #1173bb;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c28:visited,
.c28:active {
  color: #1173bb;
}

.c28:focus,
.c28:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c11 {
  box-sizing: border-box;
  width: 35px;
  height: 20px;
  padding: 0;
  border: 1px solid #ddd;
  border-radius: 5px;
  margin-right: 0.75em;
  background: none;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  line-height: 18px;
  outline: none;
}

.c11:active,
.c11:hover {
  border-color: #b7b7b7;
}

.c11:focus {
  box-shadow: 0 0 0 0.2em rgba(221,221,221,0.5);
}

.c11::-moz-focus-inner {
  border: 0;
}

.c9 {
  position: relative;
  display: inline-block;
  box-sizing: border-box;
  width: 14px;
  height: 14px;
  padding: 1px;
  border-width: 1px;
  margin: 0 0.7em;
  border-color: #000;
  border-radius: 3px;
  border-style: solid;
  font-size: 10px;
}

.c9 > img {
  position: absolute;
  top: 1px;
  left: 1px;
  width: 10px;
  height: 10px;
}

.c7 {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  white-space: nowrap;
  width: 1px;
}

.c7:focus + .c8 {
  border-color: #428bca;
  box-shadow: 0 0 0 0.2em #428bca;
}

.c5 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: hidden;
  border-color: #E6573D;
  border-style: solid;
  border-width: 1px;
}

.c5:first-child {
  border-top-left-radius: 0.5em;
  border-bottom-left-radius: 0.5em;
}

.c5:nth-last-child(2) {
  border-top-right-radius: 0.5em;
  border-bottom-right-radius: 0.5em;
}

.c12 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: hidden;
  border-color: #FAB470;
  border-style: solid;
  border-width: 1px;
}

.c12:first-child {
  border-top-left-radius: 0.5em;
  border-bottom-left-radius: 0.5em;
}

.c12:nth-last-child(2) {
  border-top-right-radius: 0.5em;
  border-bottom-right-radius: 0.5em;
}

.c14 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: hidden;
  border-color: #5E6F9E;
  border-style: solid;
  border-width: 1px;
}

.c14:first-child {
  border-top-left-radius: 0.5em;
  border-bottom-left-radius: 0.5em;
}

.c14:nth-last-child(2) {
  border-top-right-radius: 0.5em;
  border-bottom-right-radius: 0.5em;
}

.c16 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: hidden;
  border-color: #bababa;
  border-style: solid;
  border-width: 1px;
}

.c16:first-child {
  border-top-left-radius: 0.5em;
  border-bottom-left-radius: 0.5em;
}

.c16:nth-last-child(2) {
  border-top-right-radius: 0.5em;
  border-bottom-right-radius: 0.5em;
}

.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: linear-gradient(to right,rgba(230,87,61,0.5),rgba(230,87,61,0.5) 2em,rgba(0,0,0,0) 2em,rgba(0,0,0,0));
  background-repeat: no-repeat;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c13 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: linear-gradient(to right,rgba(250,180,112,0.5),rgba(250,180,112,0.5) 2em,rgba(0,0,0,0) 2em,rgba(0,0,0,0));
  background-repeat: no-repeat;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c15 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: linear-gradient(to right,rgba(94,111,158,0.5),rgba(94,111,158,0.5) 2em,rgba(0,0,0,0) 2em,rgba(0,0,0,0));
  background-repeat: no-repeat;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c17 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: linear-gradient(to right,rgba(186,186,186,0.5),rgba(186,186,186,0.5) 2em,rgba(0,0,0,0) 2em,rgba(0,0,0,0));
  background-repeat: no-repeat;
  font-size: 14px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c10 {
  padding: 0.375em 0.75em;
  line-height: 1.15em;
}

.c19 {
  box-sizing: border-box;
  height: calc(2em + 2px);
  padding: 0.375em 0.75em;
  border-color: #6c757d;
  border-style: solid;
  border-width: 1px;
  border-radius: 0.5em;
  background-color: #f8f9fa;
  cursor: pointer;
  font-size: 1em;
  line-height: 1.25;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 35px;
  height: 20px;
  padding: 0;
  border-radius: 5px;
  margin: 0 0.5em;
}

.c19:active,
.c19:hover {
  background-color: #cbd3da;
}

.c19:disabled {
  background-color: rgba(248,249,250,0.5);
  cursor: not-allowed;
}

.c19:focus {
  box-shadow: 0 0 0 0.2em rgba(108,117,125,0.5);
}

.c19 svg {
  position: relative;
  top: 0.11em;
  width: 0.9em;
  height: 0.9em;
}

.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c20 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  outline: none;
  padding: 0 3px;
  border: none;
  background: none;
  cursor: pointer;
}

.c20 img {
  position: relative;
  top: 0.13em;
  width: 14px;
  height: 14px;
  border-radius: 7px;
}

.c20:focus img {
  box-shadow: 0 0 0 0.2em rgba(70,130,180,0.5);
}

.c0 {
  padding: 0 80px 0 115px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  margin-bottom: 1em;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 0.5em;
}

.c21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

.c21 input[type='checkbox'] {
  position: relative;
  top: 2px;
}

.c21 label {
  margin-left: 1em;
}

.c21 label:first-child {
  margin-left: 0;
}

.c22 {
  margin-right: 1ch;
}

.c25 {
  box-sizing: border-box;
  height: calc(2em + 2px);
  padding: 0.375em 0.75em;
  border-color: #6c757d;
  border-style: solid;
  border-width: 1px;
  border-radius: 0.5em;
  background-color: #f8f9fa;
  cursor: pointer;
  font-size: 1em;
  line-height: 1.25;
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 35px;
  height: 20px;
  padding: 0;
  border-radius: 5px;
  line-height: 18px;
}

.c25:active,
.c25:hover {
  background-color: #cbd3da;
}

.c25:disabled {
  background-color: rgba(248,249,250,0.5);
  cursor: not-allowed;
}

.c25:focus {
  box-shadow: 0 0 0 0.2em rgba(108,117,125,0.5);
}

.c25 svg {
  position: relative;
  top: 0.11em;
  width: 0.9em;
  height: 0.9em;
}

.c27 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

@media (max-width:1200px) {
  .c3 .c4 {
    border-radius: 0.5em;
    margin: 0 0.5em 0.5em 0;
  }

  .c3 .c18 {
    margin: 0 0.5em 0.5em;
  }
}

@media (max-width:900px) {
  .c0 {
    padding: 0;
  }
}

@media (max-width:600px) {
  .c2 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
  }
}

@media (max-width:600px) {
  .c21 {
    margin-bottom: 0.5em;
  }
}

<div
    className="c0"
  >
    <div
      className="c1"
    >
      <div
        className="c2"
      >
        <div>
          <div
            className="c3"
            id="clinvar-track-included-categories"
          >
            <span
              className="c4 c5"
            >
              <label
                className="c6"
                htmlFor="clinvar-track-included-categories-pathogenic"
              >
                <input
                  checked={true}
                  className="c7"
                  id="clinvar-track-included-categories-pathogenic"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  aria-hidden={true}
                  className="c8 c9"
                >
                  <img
                    alt=""
                    src="test-file-stub"
                  />
                </span>
                <span
                  className="c10"
                >
                  Pathogenic / likely pathogenic
                </span>
              </label>
              <button
                className="c11"
                onClick={[Function]}
              >
                only
              </button>
            </span>
            <span
              className="c4 c12"
            >
              <label
                className="c13"
                htmlFor="clinvar-track-included-categories-uncertain"
              >
                <input
                  checked={true}
                  className="c7"
                  id="clinvar-track-included-categories-uncertain"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  aria-hidden={true}
                  className="c8 c9"
                >
                  <img
                    alt=""
                    src="test-file-stub"
                  />
                </span>
                <span
                  className="c10"
                >
                  Uncertain significance / conflicting
                </span>
              </label>
              <button
                className="c11"
                onClick={[Function]}
              >
                only
              </button>
            </span>
            <span
              className="c4 c14"
            >
              <label
                className="c15"
                htmlFor="clinvar-track-included-categories-benign"
              >
                <input
                  checked={true}
                  className="c7"
                  id="clinvar-track-included-categories-benign"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  aria-hidden={true}
                  className="c8 c9"
                >
                  <img
                    alt=""
                    src="test-file-stub"
                  />
                </span>
                <span
                  className="c10"
                >
                  Benign / likely benign
                </span>
              </label>
              <button
                className="c11"
                onClick={[Function]}
              >
                only
              </button>
            </span>
            <span
              className="c4 c16"
            >
              <label
                className="c17"
                htmlFor="clinvar-track-included-categories-other"
              >
                <input
                  checked={true}
                  className="c7"
                  id="clinvar-track-included-categories-other"
                  onChange={[Function]}
                  type="checkbox"
                />
                <span
                  aria-hidden={true}
                  className="c8 c9"
                >
                  <img
                    alt=""
                    src="test-file-stub"
                  />
                </span>
                <span
                  className="c10"
                >
                  Other
                </span>
              </label>
              <button
                className="c11"
                onClick={[Function]}
              >
                only
              </button>
            </span>
            <button
              className="c18 c19"
              onClick={[Function]}
              type="button"
            >
              all
            </button>
          </div>
           
          <button
            className="c20"
            onClick={[Function]}
            type="button"
          >
            <img
              alt=""
              aria-hidden="true"
              src="test-file-stub"
            />
            <span
              style={
                {
                  "border": "0",
                  "clip": "rect(0 0 0 0)",
                  "height": "1px",
                  "margin": "-1px",
                  "overflow": "hidden",
                  "padding": "0",
                  "position": "absolute",
                  "whiteSpace": "nowrap",
                  "width": "1px",
                }
              }
            >
              More information
            </span>
          </button>
        </div>
      </div>
      <div
        className="c2"
      >
        <div
          className="c21"
        >
          <div
            className="c22"
          >
            <label
              className="c23"
              htmlFor="clinvar-track-include-lof"
            >
              <input
                checked={true}
                className="c24"
                disabled={false}
                id="clinvar-track-include-lof"
                onChange={[Function]}
                type="checkbox"
              />
              pLoF
            </label>
             
            <button
              className="c25"
              onClick={[Function]}
              type="button"
            >
              only
            </button>
          </div>
          <div
            className="c22"
          >
            <label
              className="c23"
              htmlFor="clinvar-track-include-missense"
            >
              <input
                checked={true}
                className="c24"
                disabled={false}
                id="clinvar-track-include-missense"
                onChange={[Function]}
                type="checkbox"
              />
              Missense / Inframe indel
            </label>
             
            <button
              className="c25"
              onClick={[Function]}
              type="button"
            >
              only
            </button>
          </div>
          <div
            className="c22"
          >
            <label
              className="c23"
              htmlFor="clinvar-track-include-synonymous"
            >
              <input
                checked={true}
                className="c24"
                disabled={false}
                id="clinvar-track-include-synonymous"
                onChange={[Function]}
                type="checkbox"
              />
              Synonymous
            </label>
             
            <button
              className="c25"
              onClick={[Function]}
              type="button"
            >
              only
            </button>
          </div>
          <div
            className="c22"
          >
            <label
              className="c23"
              htmlFor="clinvar-track-include-other"
            >
              <input
                checked={true}
                className="c24"
                disabled={false}
                id="clinvar-track-include-other"
                onChange={[Function]}
                type="checkbox"
              />
              Other
            </label>
             
            <button
              className="c25"
              onClick={[Function]}
              type="button"
            >
              only
            </button>
          </div>
          <button
            className="c25"
            onClick={[Function]}
            style={
              {
                "marginLeft": "0.5em",
              }
            }
            type="button"
          >
            all
          </button>
        </div>
        <button
          className="c26"
          onClick={[Function]}
          style={
            {
              "flexShrink": 0,
            }
          }
          type="button"
        >
          Expand to all variants
        </button>
      </div>
      <div
        className="c27"
      >
        <label
          className="c23"
          htmlFor="clinvar-track-in-gnomad"
        >
          <input
            checked={false}
            className="c24"
            disabled={false}
            id="clinvar-track-in-gnomad"
            onChange={[Function]}
            type="checkbox"
          />
          Only show ClinVar variants that are in gnomAD
        </label>
        <label
          htmlFor="star-filtering"
        >
          Filter by
           
          <a
            className="c28"
            href="https://www.ncbi.nlm.nih.gov/clinvar/docs/review_status/"
            rel="noopener noreferrer"
            target="_blank"
          >
            review status
          </a>
          :  
          <select
            id="clinvar-star-filter"
            onChange={[Function]}
            value={0}
          >
            <option
              value={0}
            >
               0-4 Stars 
            </option>
            <option
              value={1}
            >
               
              &gt;
              =1 Stars 
            </option>
            <option
              value={2}
            >
               
              &gt;
              =2 Stars 
            </option>
            <option
              value={3}
            >
               
              &gt;
              =3 Stars 
            </option>
            <option
              value={4}
            >
               4 Stars 
            </option>
          </select>
        </label>
      </div>
    </div>
  </div>,
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  height: 100%;
  padding-right: 20px;
}

<div
    className="c0"
  >
    <div
      className="c1"
    >
      <div
        className="c2"
        style={
          {
            "width": 4,
          }
        }
      >
        <div
          className="c3"
        >
          ClinVar variants (
          5
          )
        </div>
      </div>
      <div
        className="c4"
        style={
          {
            "width": 3,
          }
        }
      >
        <svg
          height={50}
          style={
            {
              "overflow": "visible",
            }
          }
          width={3}
        >
          <g>
            <text
              dy="0.3em"
              textAnchor="end"
              x={-7}
              y={0}
            >
              1
            </text>
            <line
              stroke="#000"
              strokeWidth={1}
              x1={-5}
              x2={0}
              y1={0}
              y2={0}
            />
            <text
              dy="0.3em"
              textAnchor="end"
              x={-7}
              y={50}
            >
              0
            </text>
            <line
              stroke="#000"
              strokeWidth={1}
              x1={-5}
              x2={0}
              y1={50}
              y2={50}
            />
            <line
              stroke="#000"
              strokeWidth={1}
              x1={0}
              x2={0}
              y1={0}
              y2={50}
            />
          </g>
          <g />
          <line
            stroke="#000"
            x1={0}
            x2={3}
            y1={50}
            y2={50}
          />
        </svg>
      </div>
    </div>
  </div>,
]
`;
